### PR TITLE
Refactor sign get to allow for passing in an allData query parameter

### LINKED ISSUE #50: Add ability to pass a custom header to AWS OAuth Proxy
We need the ability to pass in a custom header so we can call JD API with that same custom header.

### PR DESCRIPTION
**Pull request by nbodesa**
This will result in us sending a No_Paging header to get request to JD's API

resolves #50

### CODE REVIEW

File: src/SignAndGet.js [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+  accessTokenSecret,
+  optionalAuthorizeCallbackUri,
+  allData
+) => {
+  const config = await getConfig();
+  const authorizeCallbackUri =
+    optionalAuthorizeCallbackUri || config.authorizeCallbackUri;
+
+  let customHeaders = { Accept: "application/vnd.deere.axiom.v3+json" };
+  if (allData != undefined) {
```
Comment (teolandon) [inline src/SignAndGet.js:17]: Probably doesn't make a difference here, but isn't `!==` considered better style?

File: app.js [hunk 2]
```diff
@@ -115,12 +120,19 @@ exports.oAuthSignRequestGet = async (event) => {
...
     accessToken,
-    accessTokenSecret
+    accessTokenSecret,
+    allData,
   } = receivedData.queryStringParameters;
 
-  const response = await doSignAndGet(url, accessToken, accessTokenSecret, getOptionalAuthorizeCallbackUri(event))
-    .then(responseData => sendResponse(event, responseData))
-    .catch(error => sendError(event, error));
+  const response = await doSignAndGet(
```
Comment (blarmon) [inline app.js:127]: It seems sorta weird to me to do both async/await and then/catch in the same space, especially on the same call. A try/catch with async/await might be a good alternative

File: app.js [hunk 3]
```diff
@@ -156,12 +163,17 @@ exports.oAuthSignRequestDelete = async (event) => {
...
     url,
     accessToken,
-    accessTokenSecret
+    accessTokenSecret,
   } = receivedData.queryStringParameters;
 
-  const response = await doSignAndDelete(url, accessToken, accessTokenSecret, getOptionalAuthorizeCallbackUri(event))
-    .then(responseData => sendResponse(event, responseData))
-    .catch(error => sendError(event, error));
+  const response = await doSignAndDelete(
```
Comment (blarmon) [inline app.js:169]: It seems sorta weird to me to do both async/await and then/catch in the same space, especially on the same call. A try/catch with async/await might be a good alternative

File: test/SignAndGet.test.js [hunk 4]
```diff
@@ -0,0 +1,290 @@
...
+
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+
+  it("gets a set of temporary OAuth tokens with same values from config", async () => {
+    const expectedFirstLegUri = chance.string();
+    const expectedThirdLegUri = chance.string();
+    const expectedClientKey = chance.string();
+    const expectedClientSecret = chance.string();
```
Comment (blarmon) [inline test/SignAndGet.test.js:57]: I think what we had decided on was moving all of this setup into the beforeEach and only having assertions within our it blocks

File: test/App.test.js [hunk 5]
```diff
@@ -70,333 +77,366 @@ describe('Lambda handlers', () => {
       expect(fakeGetOAuthRequestToken).toBeCalledWith(expect.any(Function));
     });
 
-    it('should return the request tokens if the response is successful', async () => {
-      const OAuth = require('oauth');
+    it("should return the request tokens if the response is successful", async () => {
+      const OAuth = require("oauth");
 
       const fakeError = null;
       const fakeRequestToken = chance.string();
```
Comment (blarmon) [inline test/App.test.js:84]: Same comment generally as I had re: the test setup in doSignAndGet

File: test/OAuthSignRequest.test.js [hunk 6]
```diff
@@ -21,127 +20,49 @@ describe('OAuth Sign Request', () => {
...
           callback(null, fakeResponseData, { statusCode: 200 });
         },
       }));
 
       return OAuth;
     };
 
-    it('is a function', () => {
-      const { doSignAndDelete } = require('../src/OAuthSignRequest');
+    it("is a function", () => {
```
Comment (blarmon) [inline test/OAuthSignRequest.test.js:50]: This seems to be implicitly tested elsewhere

File: test/OAuthSignRequest.test.js [hunk 7]
```diff
@@ -239,65 +176,85 @@ describe('OAuth Sign Request', () => {
...
           callback(null, fakeResponseData, { statusCode: 200 });
         },
       }));
 
       return OAuth;
     };
 
-    it('is a function', () => {
-      const { doSignAndPost } = require('../src/OAuthSignRequest');
+    it("is a function", () => {
```
Comment (blarmon) [inline test/OAuthSignRequest.test.js:242]: same comment as for doSignAndDelete

File: src/SignAndGet.js (hunk 1 above)
Comment (erik-source) [reply src/SignAndGet.js:17]: I think so? I'd be fine with that

File: test/OAuthSignRequest.test.js (hunk 6 above)
Comment (erik-source) [reply test/OAuthSignRequest.test.js:50]: Hmmm you think?

File: src/SignAndGet.js (hunk 1 above)
Comment (nbodesa) [reply src/SignAndGet.js:17]: Stricter matching, thanks forgot this is js. :)

File: app.js (hunk 3 above)
Comment (nbodesa) [reply app.js:169]: We can change this on the get endpoint for now. We don't want to impact others and would need to add more scope to cover these before changing.

File: test/OAuthSignRequest.test.js (hunk 6 above)
Comment (erik-source) [reply test/OAuthSignRequest.test.js]: Fixed

File: app.js [hunk 8]
```diff
@@ -124,15 +124,20 @@ exports.oAuthSignRequestGet = async (event) => {
...
+  try {
+    const body = await doSignAndGet(
+      url,
+      accessToken,
+      accessTokenSecret,
+      getOptionalAuthorizeCallbackUri(event),
+      allData
+    );
+
+    response = sendResponse(event, body);
```
Comment (teolandon) [inline app.js:137]: Can we just return here and on line 139 instead of storing off `response` just to return it later?

Comment (blarmon) [reply app.js:137]: b'agree

File: test/SignAndGet.test.js [hunk 9]
```diff
@@ -50,241 +50,311 @@ describe("SignAndGet", () => {
...
+        chance.string(),
+        chance.string(),
+        chance.string()
+      );
     });
 
-    await expect(
-      underTest.doSignAndGet(chance.string(), chance.string(), chance.string())
-    ).resolves.toMatch(expectedStatusText);
+    it("return an error", async () => {
```
Comment (blarmon) [inline test/SignAndGet.test.js:263]: probably "returns an error", or even better "should return an error"

File: test/SignAndGet.test.js [hunk 10]
```diff
@@ -50,241 +50,311 @@ describe("SignAndGet", () => {
...
```
Comment (blarmon) [inline test/SignAndGet.test.js:294]: same as test above

**Pull request merged**